### PR TITLE
feat: add fastify example

### PR DESCRIPTION
**Pull request by songkeys**
The code works good in my machine!

I just finished the example in /dev fold. Pretty much a direct copy paste of your work tbh.

And I noticed `reply.session` should actually be typed in the plugin itself rather than the user end?

### CODE REVIEW

File: apps/examples/fastify/src/config/auth.config.ts [hunk 1]
```diff
@@ -1,68 +1,69 @@
...
-import Passage from "@auth/fastify/providers/passage"
-import Pinterest from "@auth/fastify/providers/pinterest"
-import Reddit from "@auth/fastify/providers/reddit"
-import Slack from "@auth/fastify/providers/slack"
-import Spotify from "@auth/fastify/providers/spotify"
-import Twitch from "@auth/fastify/providers/twitch"
-import Twitter from "@auth/fastify/providers/twitter"
-import WorkOS from "@auth/fastify/providers/workos"
-import Zoom from "@auth/fastify/providers/zoom"
+import Apple from "@auth/express/providers/apple"
```
Comment (hillac) [inline apps/examples/fastify/src/config/auth.config.ts]: Why did you change this to express?

File: packages/frameworks-fastify/src/index.ts [hunk 2]
```diff
@@ -141,6 +130,12 @@ import type { FastifyRequest, FastifyPluginAsync } from "fastify"
 import formbody from "@fastify/formbody"
 import { toWebRequest, toFastifyReply } from "./lib/index.js"
 
+declare module "fastify" {
```
Comment (hillac) [inline packages/frameworks-fastify/src/index.ts]: I'll check with the maintainers but I think we probably should leave the session management up to the user given we are having them define their own auth prehandlers. https://github.com/nextauthjs/next-auth/pull/9587#issuecomment-1892621150

File: apps/dev/fastify/src/config/auth.config.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import Slack from "@auth/express/providers/slack"
+import Spotify from "@auth/express/providers/spotify"
+import Twitch from "@auth/express/providers/twitch"
+import Twitter from "@auth/express/providers/twitter"
+import WorkOS from "@auth/express/providers/workos"
+import Zoom from "@auth/express/providers/zoom"
+
+export const authConfig = {
+  trustHost: true,
+  debug: process.env.NODE_ENV !== "production" ? true : false,
```
Comment (hillac) [inline apps/dev/fastify/src/config/auth.config.ts]: unnecessary ternary
```suggestion
  debug: process.env.NODE_ENV !== "production",
```

File: apps/dev/fastify/views/protected.pug
Comment (hillac) [inline apps/dev/fastify/views/protected.pug:1]: In my app I actually split this into two files, because I wanted the page to be behind an authentication handler that won't even let you see the authenticated page if you're not authenticated. The way it's done in the express app leaves the authentication logic up to the pug template. You'll need to use my pug files since you copied my handlers.

File: packages/frameworks-fastify/src/index.ts (hunk 2 above)
Comment (songkeys) [reply packages/frameworks-fastify/src/index.ts]: Ah, I overlooked the part where users need to define the `session` decorator themselves in their prehandlers. I've reverted that in the next commit.

However, I believe many Fastify plugin libraries within its ecosystem actually define the decorators themselves. For instance, take a look at https://github.com/fastify/fastify-passport. Users can create the auth middleware using a helper from the library:

```ts
server.get(
  '/',
  { preValidation: fastifyPassport.authenticate('test', { authInfo: false }) },
  async () => 'hello world!'
)
```

That said, this could result in duplicated names for decorator properties: https://github.com/fastify/fastify/issues/1955 . But I suppose that's just how things work in the Fastify ecosystem. We can stick to the current design though.

**Comment by songkeys** [comment]
@hillac Hi! Could you please take another look? I've corrected the issues. Some mistakes are made because I copied files for the whole auth config from the express folder without double-checking after completing testing.

File: packages/frameworks-fastify/src/index.ts (hunk 2 above)
Comment (hillac) [reply packages/frameworks-fastify/src/index.ts]: https://github.com/nextauthjs/next-auth/pull/9587#issuecomment-1892166589 Yeah I agree it would be simpler for users but they want to keep it flexible which is also fair.
